common/v2/database/v1.0.0/generateDefaultValues.ts

```diff
@@ -0,0 +1,185 @@
+/* Transducers */
+const addNetworks = add(LSKeys.NETWORKS)((networks: SeedData) => {
+  const formatNetwork = (n: NetworkLegacy): Network => {
+    const baseAssetUuid = generateUUID();
+    return {
+      // Also availbale are: blockExplorer, tokenExplorer, tokens aka assets, contracts
+      id: n.id,
+      name: n.name,
+      chainId: n.chainId,
+      isCustom: n.isCustom,
+      isTestnet: n.isTestnet,
+      color: n.color,
+      gasPriceSettings: n.gasPriceSettings,
+      shouldEstimateGasPrice: n.shouldEstimateGasPrice,
+      dPaths: {
+        ...n.dPaths,
+        default: n.dPaths[WalletId.MNEMONIC_PHRASE] // Set default dPath
+      },
+      blockExplorer: n.blockExplorer,
+      tokenExplorer: n.tokenExplorer,
+      contracts: [],
+      assets: [],
+      baseAsset: baseAssetUuid, // Set baseAssetUuid
+      baseUnit: n.unit,
+      nodes: NODES_CONFIG[n.name as NetworkId]
+    };
+  };
+
+  return R.mapObjIndexed(formatNetwork, networks);
+});
+
+const addContracts = add(LSKeys.CONTRACTS)(
+  (networks: Record<NetworkId, NetworkLegacy>, store: LocalStorage) => {
+    const formatContract = (id: NetworkId) => (c: ContractLegacy): ExtendedContract => ({
+      uuid: generateContractUUID(c.abi),
+      name: c.name,
+      address: c.address,
+      abi: c.abi,
+      networkId: id
+    });
+
+    // Transform { ETH: { contracts: [ {<contract>} ] }}
+    // to   { <contract_uuid>: {<contract>} }
+    return R.pipe(
+      R.map(({ id, contracts }) => ({ id, contracts })),
+      R.filter(({ contracts }) => contracts),
+      R.chain(({ id, contracts }): ExtendedAsset[] => contracts.map(formatContract(id))),
+      R.reduce(toObject('uuid'), {}),
+      R.mergeRight(store.contracts)
+    )(toArray(networks));
+  }
+);
+
+const addContractsToNetworks = add(LSKeys.NETWORKS)((_, store: LocalStorage) => {
+  const getNetworkContracts = (n: Network) => {
+    const nContracts = R.filter((c: ExtendedContract) => c.networkId === n.id, store.contracts);
+    return {
+      ...n,
+      contracts: toArray(nContracts).map(c => c.uuid)
+    };
+  };
+  return R.mapObjIndexed(getNetworkContracts, store.networks);
+});
+
+const addBaseAssetsToAssets = add(LSKeys.ASSETS)((_, store: LocalStorage) => {
+  const formatAsset = (n: Network): Asset => ({
+    uuid: n.baseAsset,
+    ticker: n.baseUnit,
+    name: n.name,
+    networkId: n.id,
+    type: 'base',
+    decimal: DEFAULT_ASSET_DECIMAL
+  });
+
+  // From { <networkId>: { baseAsset: <asset_uui> } }
+  // To   { <asset_uuid>: <asset> }
+  return R.pipe(
+    toArray,
+    R.map(formatAsset),
+    R.reduce((acc, curr) => ({ ...acc, [curr.uuid]: curr }), {}),
+    R.mergeRight(store.assets) // Ensure we return an object with existing assets as well
+  )(store.networks);
+});
+
+const addFiatsToAssets = add(LSKeys.ASSETS)((fiats: Fiat[], store: LocalStorage) => {
+  const formatFiat = ({ code, name }: Fiat): ExtendedAsset => ({
+    uuid: generateUUID(),
```

Comment:
Should we make these deterministic as well? Maybe generate uuid from the fiat code? Those are unique afaik.

Reply:
changed to `generateAssetUUID(code, name)`